core/shared/src/main/scala/zio/ZIOApp.scala

```diff
@@ -90,13 +90,12 @@ trait ZIOApp extends ZIOAppPlatformSpecific with ZIOAppVersionSpecific { self =>
           layer +!+ ZLayer.environment[ZEnv with ZIOAppArgs]
 
       for {
-        _          <- installSignalHandlers
-        newRuntime <- ZIO.runtime[ZEnv].map(_.mapRuntimeConfig(hook))
-        result     <- newRuntime.run(run.provideLayer(newLayer))
+        _      <- installSignalHandlers
+        result <- run.provideLayer(newLayer)
       } yield result
     }
 
-  def runtime: Runtime[ZEnv] = Runtime.default
+  def runtime: Runtime[ZEnv] = Runtime.default.mapRuntimeConfig(hook)
```

Comment:
By this we risk not applying the runtime config if someone overrides `runtime`

Reply:
Do you think it would be better to add `final def runtimeWithHook = runtime.mapRuntimeConfig(hook)` in ZIOApp and then call this method instead from the different main methods? If yes, do you have a better name for the new final method?

Reply:
Can't we just create a new runtime internally like we currently do?

Reply:
We previously had this line:

```scala
newRuntime <- ZIO.runtime[ZEnv].map(_.mapRuntimeConfig(hook))
```

Reply:
If we have it there we need to do `newRuntime.run` in the next step and that was the problematic code.
Maybe the interruption can be propagated in some other way that I don't understand.

Reply:
Will take a look in a few hours.

Reply:
You can just do `run.provideLayer(newLayer) @@ ZIOAspect.runtimeConfig(hook)` using a `ZIOAspect`.

Reply:
Oh nice!
That was much cleaner!

Reply:
Anything else I need to change?

Reply:
Before the magic "log message" came from FiberContext.unsafeReportUnhandled when the inner runtime.run ended with a failure.

The message was: `Fiber Runtime(66,1646676692,zio.ZIOAppSpec.spec(ZIOAppSpec.scala:65)) did not handle an error`
The cause was 
```
Exception in thread "zio-fiber-66" java.lang.String: Uh oh!
	at zio.ZIOAppSpec.spec.app1(ZIOAppSpec.scala:62)
	at zio.ZIOAppSpec.spec(ZIOAppSpec.scala:65)```

Since this doesn't longer happen inside the logging aspect, it should be fine, I think!

Reply:
Right, but then a logger was installed, a failure occurred, and the logger was not called. I think the underlying problem here is that `run` is not propagating interruption. We can address this by updating the implementation of `Runtime.run` as follows:

```scala
final def run[E, A](zio: ZIO[R, E, A])(implicit trace: ZTraceElement): IO[E, A] =
  ZIO.fiberId.flatMap { fiberId =>
    IO.asyncInterrupt[E, A] { callback =>
      val canceler = unsafeRunAsyncCancelable(zio)(exit => callback(ZIO.done(exit)))
      Left(ZIO.succeed(canceler(fiberId)))
    }
  }
```

Then we can revert all the changes to `ZIOApp` and all the changes to the tests other than your new test case.

Reply:
OK, fine! The code is updated.

Actually my first try to solve the issue was something like that, but I didn't have the skill to complete it... :-)